module/VuFind/src/VuFind/RecordDriver/EDS.php

```diff
@@ -567,7 +607,235 @@ protected function getSupportedCitationFormats()
      */
     public function exportDisabled($format)
     {
-        // EDS is not export-friendly; disable all formats.
-        return true;
+        // at the moment we only support RIS
+        return strtolower($format) != 'ris';
+    }
+    
+     /**
+     * This method can fetch fields out of the deeply nested RecordInfo array. 
+     * many fields are here in a more readable and atomic format
+     * 
+     * The latter two params are only for internal use (recursion)
+     *  
+     * @param array $arrayKeys Key path to the needed value
+     * @param int   $level     only used for recursion
+     * @param array $fields    only used for recursion
+      * 
+     * @return array|string
+     */
+    protected function getFieldRecursive($arrayKeys, $level = 0, $fields = null)
+    {
+        // special case: we have to iterate through items array
```

Comment:
This is really two totally separate functions in one. I wonder if we could make this more clear by making the inputs more distinct. For example, what if instead of taking an array as input, we took a string, and it could be of one of two different formats:

"Items:ItemField"

or

"Fields:Path/To/Value/We/Want"

Then we could explode on the colon and use a switch statement to execute the appropriate logic, with the recursive logic living in a different method entirely. I think this might also lead to a more readable prioritizedFields property.

What do you think? I'm willing to make the proposed changes if you think it's a good idea but don't have time. If you think it's better as-is, we can leave it alone, but it might still be worth thinking about some more minor changes to improve readability and flow.